parseSheet.py

```diff
@@ -63,9 +63,11 @@ def insertSpreadsheetDict(d,owner):
 
 
 def importSpreadsheet(infilename,owner):
-  d = parseSpreadsheet(infilename)
-  insertSpreadsheetDict(d,owner)
-
+  if(infilename != ""):
```

Comment:
this should handle more than just an explicit empty string from a cancel - it should also handle empty or whitespace entry as well. see the best answer for https://stackoverflow.com/questions/9573244/how-to-check-if-the-string-is-empty

Reply:
you have a very good point, and are getting towards the best solution.

since the goal is either good data from the spreadsheet or asking the user to select a valid file, you should probably check for not just an OSError for a missing/unreadable file, but also for badly formatted file.

I would suggest checking offline to see what exception that throws.

minimize the types caught to prevent other issues to prevent other issues from being missed

Reply:
changes look good! already approved, so resolving.